mediapipe/tasks/ios/audio/core/sources/MPPAudioTaskRunner.mm

```diff
@@ -127,6 +131,70 @@ - (nullable instancetype)initWithTaskInfo:(MPPTaskInfo *)taskInfo
                                     : std::nullopt;
 }
 
+- (BOOL)processStreamAudioClip:(MPPAudioData *)audioClip
+       timestampInMilliseconds:(NSInteger)timestampInMilliseconds
+                         error:(NSError **)error {
+  if (_runningMode != MPPAudioRunningModeAudioStream) {
+    [MPPCommonUtils
+        createCustomError:error
+                 withCode:MPPTasksErrorCodeInvalidArgumentError
+              description:[NSString stringWithFormat:@"The audio task is not initialized with "
+                                                     @"audio stream mode. Current Running Mode: %@",
+                                                     MPPAudioRunningModeDisplayName(_runningMode)]];
+    return NO;
+  }
+
+  if (![self checkOrSetSampleRate:audioClip.format.sampleRate error:error]) {
+    return NO;
+  }
+
+  Packet matrixPacket = [MPPAudioPacketCreator createPacketWithAudioData:audioClip
+                                                 timestampInMilliseconds:timestampInMilliseconds
+                                                                   error:error];
+  if (matrixPacket.IsEmpty()) {
+    return NO;
+  }
+
+  PacketMap inputPacketMap = {{_audioInputStreamName.cppString, matrixPacket}};
+  return [self sendPacketMap:inputPacketMap error:error];
+}
+
+- (BOOL)checkOrSetSampleRate:(double)sampleRate error:(NSError **)error {
+  if (_runningMode != MPPAudioRunningModeAudioStream) {
+    [MPPCommonUtils
+        createCustomError:error
+                 withCode:MPPTasksErrorCodeInvalidArgumentError
+              description:[NSString stringWithFormat:@"The audio task is not initialized with "
+                                                     @"audio stream mode. Current Running Mode: %@",
+                                                     MPPAudioRunningModeDisplayName(_runningMode)]];
+    return NO;
+  }
+
+  if (_sampleRate <= 0) {
```

Comment:
Maybe add comment that this means the value hasn't been initialized.

Reply:
I'd suggest comparing to the default value instead for clarity.